components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```diff
@@ -0,0 +1,341 @@
+        loadMockDatabaseUtil();
+        Action deactivatedAction = daoImpl.deactivateAction(action.getType().getActionType(), action.getId(),TENANT_ID);
+        loadMockDatabaseUtil();
+        Action result = daoImpl.activateAction(deactivatedAction.getType().getActionType(),deactivatedAction.getId(),TENANT_ID);
+        Assert.assertEquals(Action.Status.ACTIVE, result.getStatus());
+
+    }
+
+    @Test
+    public void testGetActionByActionId() throws ActionMgtException {
+        Action action = getAction("274274");
+        daoImpl.addAction(action.getType().getActionType(), action.getId(), action, TENANT_ID);
+        loadMockDatabaseUtil();
+        Action result = daoImpl.getActionByActionId(action.getId(),TENANT_ID);
+        Assert.assertEquals(action.getName(),result.getName());
+        Assert.assertEquals(action.getDescription(),result.getDescription());
+        Assert.assertEquals(action.getType(),result.getType());
+        Assert.assertEquals(action.getStatus(),result.getStatus());
+        Assert.assertEquals(action.getEndpoint().getUri(),result.getEndpoint().getUri());
+        Assert.assertEquals(action.getEndpoint().getAuthentication().getType(), result.getEndpoint().getAuthentication().getType());
+
+    }
+
+    @Test
+    public void testUpdateActionEndpointAuthProperties() throws ActionMgtException {
+        Action action = getAction("239239");
+        daoImpl.addAction(action.getType().getActionType(), action.getId(), action, TENANT_ID);
+        loadMockDatabaseUtil();
+        Action result = daoImpl.updateActionEndpointAuthProperties(action.getId(),authType,TENANT_ID);
+        Assert.assertEquals(2,result.getEndpoint().getAuthentication().getProperties().size());
+
+    }
+
+    @Test
+    public void testUpdateActionEndpoint() throws ActionMgtException{
+        Action action = getAction("298298");
+        daoImpl.addAction(action.getType().getActionType(), action.getId(), action, TENANT_ID);
+        loadMockDatabaseUtil();
+        EndpointConfig endpointConfig = new EndpointConfig.EndpointConfigBuilder()
+                .uri("https://example123.com")
+                .authentication(new AuthType.AuthTypeBuilder()
+                        .type(AuthType.AuthenticationType.BEARER)
+                        .properties(Arrays.asList(
+                                new AuthProperty.AuthPropertyBuilder().name("accesstoken").value("123456").isConfidential(true).build()))
+                        .build())
+                .build();
+        Action result = daoImpl.updateActionEndpoint(action.getType().getActionType(),action.getId(), endpointConfig,action.getEndpoint()
+                .getAuthentication(), TENANT_ID);
+        Assert.assertNotEquals(action.getEndpoint().getUri(),result.getEndpoint().getUri());
+        Assert.assertEquals(AuthType.AuthenticationType.BEARER, result.getEndpoint().getAuthentication().getType());
```

Comment:
check this test in this method as well.
Updating endpoint uri, and auth properties without updating authentication type